jessenovotny/email_relay_service
# Email Relay Service

This change will introduce an email relay service to the application. 

## Requirements:
- users and groups are managed through a simple UI
- users can be added to and removed from groups
- when an email is sent to the group, it should be forwarded to all members of the group

## Implementation:
- ActionMailbox ingests new emails, triggering GroupMailbox#process
- ActionMailer used to construct and send forwarded email

## Testing:
```
bundle install
rails server
```
- navigate to http://localhost:3000/admin
- create a group and add a user
- navigate to http://localhost:3000/rails/conductor/action_mailbox/inbound_emails
- send inbound email to the group email address you created
- navigate to http://localhost:1080
- confirm that the email was forwarded to the user you created and is sent from admin@mailprotector.com

## Additional Considerations:
- automated unit tests
- security filters
- pros/cons after_commit
- restrict access to the admin interface
- concerns with persisting every email
- add indexes to optimize queries
- exception handling/reporting
- rescue/retry mechanisms
- performance tracking
- relay status tracking
- most importantly, appropriate naming
